### PR TITLE
Add onPostCreate event for activity only

### PR DESCRIPTION
**Pull request by elsennov**

### CODE REVIEW

**Comment by dlew** [comment]
I'm a bit hesitant to add this since the docs say you should rarely use it. What's the use case for `onPostCreate()` vs. `onCreate()`, especially since the listener is typically fired post-super for `onCreate()`?

**Comment by elsennov** [comment]
We need it for navigation drawer as Google suggested here
https://developer.android.com/training/implementing-navigation/nav-drawer.html

**Comment by dlew** [comment]
I looked into it and `onPostCreate()` is far more intriguing than I originally gave it credit. It's not called after `onCreate()`, it's called after `onStart()` and `onRestoreInstanceState()`. This is acceptable.

File: navi/src/main/java/com/trello/navi/internal/HandledEvents.java [hunk 1]
```diff
@@ -13,6 +13,7 @@
       Arrays.asList(
           Event.CREATE,
           Event.CREATE_PERSISTABLE,
+          Event.POST_CREATE,
```
Comment (dlew) [inline navi/src/main/java/com/trello/navi/internal/HandledEvents.java]: Nit: For correctness this should go after `Event.START`.

File: navi/src/main/java/com/trello/navi/Event.java [hunk 2]
```diff
@@ -47,6 +47,11 @@
   public static final Event<BundleBundle> CREATE_PERSISTABLE =
       new Event<>(Type.CREATE_PERSISTABLE, BundleBundle.class);
 
+    /**
+     * Emits {@link Activity#onPostCreate(Bundle)}. Emitted after super().
+     */
+    public static final Event<Bundle> POST_CREATE = new Event<>(Type.POST_CREATE, Bundle.class);
```
Comment (dlew) [inline navi/src/main/java/com/trello/navi/Event.java]: We should also have a `POST_CREATE_PERSISTABLE` for API 21+.

File: navi/src/test/java/com/trello/navi/NaviActivityTest.java [hunk 3]
```diff
@@ -56,6 +56,19 @@
     verifyNoMoreInteractions(persistableListener);
   }
 
+  @Test public void postCreateListener() {
```
Comment (dlew) [inline navi/src/test/java/com/trello/navi/NaviActivityTest.java]: Nit: Should go after tests for start.

File: navi/src/test/java/com/trello/navi/rx/RxNaviActivityTest.java [hunk 4]
```diff
@@ -52,6 +52,22 @@
     testSubscriber.assertUnsubscribed();
   }
 
+  @Test public void observePostCreate() {
```
Comment (dlew) [inline navi/src/test/java/com/trello/navi/rx/RxNaviActivityTest.java]: Nit: Should go after tests for start.

**Comment by elsennov** [comment]
I have updated my PR @dlew based on your comments

**Comment by dlew** [comment]
Great, thanks for the work!

**Comment by dlew** [comment]
(I'll look into making another release sometime next week so you can use this stuff.)

**Comment by elsennov** [comment]
Alright, you are welcome! Can't wait for the next release! :D

**Pull request merged**